File: src/crawlers/lib/tojson.py
import os
import json
import uuid

def exportJson(
  data: any,
  id: str,
  location: str,
  circuit: str,
  session_name: str,
  data_type: str,
  exportname: str,
  year: int,
):
  final_data = {
    "id": str(uuid.uuid4()),
    "weekendid": id,
    "year": year,
    "location": location,
    "circuit": circuit,
    "session_name": session_name,
    "session_type": data_type,
    "results": data,
  }
  
  # Convert data to json
  results_json = json.dumps(final_data, indent=4)
  
  path = f"../data/{year}/{location}"
  filename = f"{exportname}.json"
  
  # Create folders if needed
  os.makedirs(path, exist_ok=True)
  
  # Writes data in json file
  with open(os.path.join(path, filename), "w") as file:
    file.write(results_json)

File: src/jscrawlers/lib/tojson.ts
import * as fs from 'fs';
import * as path from 'path';
import { v4 as uuidv4 } from 'uuid';

interface ExportJsonParams {
  data: any;
  id: string;
  location: string;
  circuit: string;
  sessionName: string;
  dataType: string;
  exportName: string;
  year: number;
}

export function exportJson({
  data,
  id,
  location,
  circuit,
  sessionName,
  dataType,
  exportName,
  year
}: ExportJsonParams): void {
  const finalData = {
    id: uuidv4(),
    weekendid: id,
    year: year,
    location: location,
    circuit: circuit,
    session_name: sessionName,
    session_type: dataType,
    results: data,
  };

  // Convert data to json
  const resultsJson = JSON.stringify(finalData, null, 4);

  const dirPath = path.join(__dirname, `../data/${year}/${location}`);
  const filePath = path.join(dirPath, `${exportName}.json`);

  // Create folders if needed
  fs.mkdirSync(dirPath, { recursive: true });

  // Write data to json file
  fs.writeFileSync(filePath, resultsJson, 'utf8');
}
